shogun-gs-interceptor/src/main/java/de/terrestris/shogun/interceptor/security/access/entity/InterceptorRulePermissionEvaluator.java

```diff
@@ -0,0 +1,40 @@
+package de.terrestris.shogun.interceptor.security.access.entity;
+
+import de.terrestris.shogun.interceptor.model.InterceptorRule;
+import de.terrestris.shogun.lib.enumeration.PermissionType;
+import de.terrestris.shogun.lib.model.User;
+import de.terrestris.shogun.lib.security.SecurityContextUtil;
+import de.terrestris.shogun.lib.security.access.entity.BaseEntityPermissionEvaluator;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+/**
+ * Permission evaluator for {@link InterceptorRule}s
+ */
+@Component
+public class InterceptorRulePermissionEvaluator extends BaseEntityPermissionEvaluator<InterceptorRule> {
+
+    @Autowired
+    protected SecurityContextUtil securityContextUtil;
+
+    @Override
+    public boolean hasPermission(User user, InterceptorRule entity, PermissionType permission) {
+        List<GrantedAuthority> authorities = securityContextUtil.getGrantedAuthorities();
+        boolean isInterceptorAdmin = authorities.stream().anyMatch(
```

Comment:
maybe we can write a util for this, which can check roles for a given authorities collection
basically hasRole() or hasAnyRole() as a directly invokable method

or [isUserInRole](https://docs.spring.io/spring-security/site/apidocs/org/springframework/security/web/servletapi/SecurityContextHolderAwareRequestWrapper.html#isUserInRole(java.lang.String)) can be used, but not sure if that works

Reply:
Good idea. I'll introduce this in an upcoming PR